src/components/SkyStateGenerator.cpp

```diff
@@ -183,12 +183,12 @@ const std::shared_ptr<SkyState> SkyStateGenerator::update(float dayRatio)
   return mCurrentSkyState;
 }
 
-std::tuple<unsigned int, unsigned int> SkyStateGenerator::findPresetIndices() const
+std::tuple<std::uint32_t, std::uint32_t> SkyStateGenerator::findPresetIndices() const
 {
-  unsigned int p1i = static_cast<unsigned int>(mSkyStatePresets.size()) - 1;
-  unsigned int p2i = 0;
+  std::uint32_t p1i = static_cast<std::uint32_t>(mSkyStatePresets.size()) - 1;
```

Comment:
Shouldn't the `- 1` be part of the cast? Here, a simple C-Style cast is fine too, IMO

Reply:
I pulled it in the cast, but I personally never use C stuff when I write C++. I can still change it if you want?

Reply:
> Here, a simple C-Style cast is fine too, IMO

Or use
```
std::uint32_t(mSkyStatePresets.size() - 1)
```
which is as expressive as the C-style cast but not C.

Reply:
IMO, C++ style casts don't help in making the code more readable when casting between plain integer types. In fact, I mean, they are more verbose for a good reason, like when you're casting between pointer types or (god beware) casting away a const. They also solve the issue of `(uint32_t)value - 1` not being clear, which is nice.

But in this particular case we really just want to say, *Look, I have that number and it might be 64 bit but I want it to be 32*. There is no "order of operations", no pointer trickery, no consts. 

Can't we just use `auto` or `size_t` here to get rid of the casts all together?

Reply:
I mean `std::uint32_t(mSkyStatePresets.size() - 1)` is C++ and is as expressive as `(std::uint32_t)(mSkyStatePresets.size() - 1)`

Reply:
@mworchel This is a function style cast and is [exactly the same thing](https://stackoverflow.com/a/3487099) as a C-style cast

Reply:
Yes, I know but it's not C syntax. Whatever.

Reply:
Oh, for some reason I completely missed this discussion, sorry for that!

I didn't even know those "function style casts" worked for numbers, this is awesome!